packages/web-forms/src/demo/composables/xlsform-online.ts

```diff
@@ -0,0 +1,44 @@
+import type { Ref } from 'vue';
+
+interface XlsOnlineResponse {
+	error?: string;
+	warnings?: string[];
+	xform_url?: string;
+	itemsets_url?: string;
+}
+
+export const useXlsFormOnline = (baseUrl: Ref<string | undefined>) => {
+	const convertXlsForm = async (file: File) => {
+		let error: string | null = null;
+		let data: XlsOnlineResponse | null = null;
+
+		if (!baseUrl.value) {
+			error =
+				'Failed to initialize XLSForm Online service, please refresh the page. If problem persists, please report it on ODK Forum.';
+			return { data, error };
+		}
+
+		const formData = new FormData();
+
+		formData.append('file', file);
+
+		const response = await fetch(`${baseUrl.value}/api/xlsform`, {
+			method: 'POST',
+			body: formData,
+		})
+			.then((res) => res.json())
+			.then((json: XlsOnlineResponse) => json)
+			.catch((fetchError: string) => {
+				error = fetchError;
```

Comment:
For future iteration: there are whole classes of errors where we just spit out `FetchError`, and we could message more clearly what went wrong based on the actual kind of error. Top of mind (not exhaustive) from local testing:

- Failure to reach server at all
- CORS failure
- Empty response (saw this while trying with `pyxform-http`, before I realized I needed to use `xlsform-online`)
- 404 (and maybe other common HTTP errors, or known errors if we can enumerate them)